internal/pkg/agent/application/monitoring/stats.go

```diff
@@ -12,6 +12,39 @@ import (
 	"github.com/elastic/elastic-agent-libs/monitoring"
 )
 
+const formValueKey = "failon"
+
+type LivenessFailConfig struct {
+	Degraded bool `yaml:"degraded" config:"degraded"`
+	Failed   bool `yaml:"failed" config:"failed"`
+}
+
+// process the form values we get via HTTP
+func handleFormValues(req *http.Request) (LivenessFailConfig, error) {
```

Comment:
This would make more sense if it lived in liveness.go since that's the only place it is called.